tests/functional/test_webhook_notification.py

```diff
@@ -159,22 +173,39 @@ def test_notify_method(mock_method2, mock_method1, slack_notification_plugin):
         slack_notification_plugin.notify()
         assert slack_notification_plugin.body == str({'text': 'teflo_notification'})
 
+
     @staticmethod
     @mock.patch('httplib2.Http.request')
-    def test_notify_method_with_user_template(mock_method, scenario_resource, gchat_params):
+    def test_notify_method_with_user_template(mock_method, gc_notification_plugin):
         """To test notify method with no body and user template is provided"""
         os.system('cp ../assets/user_temp.jinja /tmp/')
         mock_method.return_value = ({'status': '200'},)
-        setattr(scenario_resource, 'passed_tasks', ['provision'])
-        setattr(scenario_resource, 'failed_tasks', [])
-        setattr(scenario_resource, 'overall_status', 0)
-        note = Notification(name='gchat1', parameters=gchat_params, config=getattr(scenario_resource, 'config'))
-        scenario_resource.add_notifications(note)
-        note.scenario = scenario_resource
-        gc_plugin = GchatNotificationPlugin(note)
-        gc_plugin.notify()
-        assert gc_plugin.body == '{"text": "hello"}'
+        gc_notification_plugin.notify()
+        assert gc_notification_plugin.body == '{"text": "hello"}'
         os.system('rm /tmp/user_temp.jinja')
 
+    @staticmethod
+    def test_get_message_headers(gc_notification_plugin):
+        """To test when no custom headers or username/pass are provided message_headers is created correctly"""
+        op = gc_notification_plugin.get_message_headers()
+        assert isinstance(op, dict)
+        assert 'Authorization' not in op.keys()
+        assert len(op) == 1
+
 
+    @staticmethod
+    def test_get_message_headers_1(webhook_notification_plugin):
+        """To test when custom headers or username/pass are provided message_headers is created correctly"""
+        op = webhook_notification_plugin.get_message_headers()
+        assert isinstance(op, dict)
+        assert 'tenant' in op.keys()
+        assert 'Authorization' in op.keys()
 
+    @staticmethod
+    def test_get_message_headers_2(webhook_notification_plugin):
```

Comment:
Maybe we could change this to a more intuitive name?

Reply:
done